packages/modular-scripts/src/convert.ts

```diff
@@ -0,0 +1,179 @@
+import { IncludeDefinition as TSConfig } from '@schemastore/tsconfig';
+import { Dependency } from '@schemastore/package';
+import execa from 'execa';
+import stripAnsi from 'strip-ansi';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import {
+  ModularPackageJson,
+  isValidModularRootPackageJson,
+} from './utils/isModularType';
+import * as logger from './utils/logger';
+import { check } from './check';
+import rimraf from 'rimraf';
+
+function cleanGit(cwd: string): boolean {
+  const trackedChanged = stripAnsi(
+    execa.sync('git', ['status', '-s'], {
+      all: true,
+      reject: false,
+      cwd,
+      cleanup: true,
+    }).stdout,
+  );
+  return trackedChanged.length === 0;
+}
+
+function resetChanges(): void {
+  execa.sync('git', ['stash', '-u']);
+  throw new Error('Failed to perform action cleanly. Stashing git changes...');
+}
+
+process.on('SIGINT', () => {
+  resetChanges();
+});
+
+export async function convert(cwd: string = process.cwd()): Promise<void> {
+  if (!cleanGit(cwd)) {
+    throw new Error(
+      'You have unsaved changes. Please save or stash them before we attempt to convert this react app to modular app.',
+    );
+  }
+
+  try {
+    if (
+      !isValidModularRootPackageJson(cwd) ||
+      !fs.existsSync(path.join(cwd, 'packages'))
+    ) {
+      const { initModularFolder } = await import('./init');
+      logger.debug(
+        "You don't have a modular repo initialized. Setting it up for you now.",
+      );
+      await initModularFolder(cwd, true);
+    }
+
+    const rootPackageJson = (await fs.readJson(
+      path.join(cwd, 'package.json'),
+    )) as ModularPackageJson;
+
+    const packageName = rootPackageJson.name as string;
+
+    logger.log(`Setting up new modular app for ${packageName}`);
+
+    // Create a modular app package folder
+    const packageTypePath = path.join(__dirname, '../types', 'app');
+    const newPackagePath = path.join(cwd, 'packages', packageName);
+    fs.mkdirpSync(newPackagePath);
+
+    const newPackageJson = fs.readJsonSync(
+      path.join(packageTypePath, 'packagejson'),
+    ) as ModularPackageJson;
+
+    // Bring key props from root package.json to new app
+    newPackageJson.name = packageName;
+    newPackageJson.modular = {
```

Comment:
@cangarugula I added this - otherwise you won't be able to start the app!

Reply:
This is already there. We read in the app type packagejson as a JSON (line 68-70) and we update it before writing it again with the updated name and browserlist.

Reply:
Ah whoops - okay I missed this.